### PR TITLE
fix:[DPMMA-2213] fix chart focus view statistics

### PR DESCRIPTION
**Pull request by AlanWierzchonCA**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Yes]
| New feature/enhancement? (use the a.x branch)      | [ No]
| Deprecations?                          | [ No]
| BC breaks? (use the c.x branch)        | [No ]
| Automated tests included?              | [Yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Fix issue: #13583
Incorrect Statistics Counting in Focus Items Chart 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))

2. Display the Landing page with Focus Item.
3. On the views chart: 1 (or increased by 1).
4. Click on a link in the focus item.
5. On the chart, both clicks and views are increased by 1 (clicks: 1, views: 2).
6. Click on the remaining 2 links. On the chart, clicks: 3, views: 4.

Expected Outcome:
Clicking on a link in the focus item should not increase the number of views.

### CODE REVIEW

File: plugins/MauticFocusBundle/Model/FocusModel.php [hunk 1]
```diff
@@ -363,12 +363,12 @@ protected function dispatchEvent($action, &$entity, $isNew = false, Event $event
     /**
      * @param bool $canViewOthers
      */
-    public function getStats(Focus $focus, $unit, \DateTime $dateFrom = null, \DateTime $dateTo = null, $dateFormat = null, $canViewOthers = true): array
+    public function getStats(Focus $focus, $unit, \DateTime $dateFrom, \DateTime $dateTo, $dateFormat = null, $canViewOthers = true): array
```
Comment (patrykgruszka) [inline plugins/MauticFocusBundle/Model/FocusModel.php:366]: This method signature change looks like BC to me. Is this required to fix this problem?

**Review by patrykgruszka — CHANGES_REQUESTED** [review]
PR fixes the issue with incorrect numbers in Focus chart. If the method signature change is not required and can be reverted, then it's good to go.

image

File: plugins/MauticFocusBundle/Model/FocusModel.php (hunk 1 above)
Comment (AlanWierzchonCA) [reply plugins/MauticFocusBundle/Model/FocusModel.php:366]: the getStats function of the FocusModel class creates a ChartQuery object on line 369, whose parameters $dateFrom, $dateTo cannot have null values.

When you use GetStats with null $dateFrom and $dateTo variables, you get a typing error from the ChartQuery constructor.

That's why I changed it.

I added a unittest: testGetStatsWithNullDates in FocusModelTest.php
You can check this with the previous getStats declaration.

Comment (patrykgruszka) [reply plugins/MauticFocusBundle/Model/FocusModel.php:366]: Makes sense, thanks for explaining

**Review by patrykgruszka — APPROVED** [review]

File: plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\PageBundle\Entity\Hit;
+use MauticPlugin\MauticFocusBundle\Entity\Focus;
+use MauticPlugin\MauticFocusBundle\Entity\Stat;
+use MauticPlugin\MauticFocusBundle\Model\FocusModel;
+
+/**
+ * @property object|null $focusModel
+ */
```
Comment (escopecz) [inline plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php]:
```suggestion
```
I don't see any reason for this comment

File: plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
+<?php
+
```
Comment (escopecz) [inline plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php:2]:
```suggestion

declare(strict_types=1);

```

**Review by escopecz — CHANGES_REQUESTED** [review]

File: plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php (hunk 2 above)
Comment (AlanWierzchonCA) [reply plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php]: I removed this.

File: plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php (hunk 3 above)
Comment (AlanWierzchonCA) [reply plugins/MauticFocusBundle/Tests/Model/FocusModelFunctionalTest.php:2]: I added this.

**Review by escopecz — APPROVED** [review]
Looks good codewise 👍

**Pull request merged**